### DISCUSSION TITLE
We should be less aggressive on disposing our controls

### DISCUSSION THREAD

**Discussion by pictos** [CommunityToolkit/Maui#873] [General]
### The "Problem"

On our toolkit, we followed the pattern, also promoted by me, of disposing of Controls ASAP, at first glance that looks like the correct approach to follow and that was the Xamarin.Forms (XF) way to do it and that has its issues.

[On .NET MAUI this behavior has changed](https://github.com/dotnet/maui/issues/3604#issuecomment-1132226021), so now the framework doesn't call `Dispose` or `DisconnectHandler` on controls ASAP, the clean-up is the user's (dev) responsibility. And the reason for that is:

> There's no way that framework can guess when the user will use the control again.

So for example, let's say that our user has a MediaPlayer application and uses our `MediaElement` control and a Player page is created when the user selects a media to run. Now imagine the user navigates to another page inside the application, from our current implementation the `MediaElement` will be disposed of, meaning the `MediaElement` control (shared and platform-specific areas) will be destroyed. With that, the media will stop playing. Can you see the issue?

Probably the desired behavior is to have the player still alive and playing the media. But with the current pattern that we have that isn't possible. (For a more complete discussion around this, please see this [discussion](https://github.com/dotnet/maui/pull/7349#discussion_r878385964).

Another situation that could happen, is the user wants to cache a Popup, that isn't possible today due to our pattern, as you can see [here](https://github.com/CommunityToolkit/Maui/discussions/700).

Another issue that I see during our implementation is to implement `destructors` on classes which isn't very safe... And, IMHO, if the .NET MAUI (a way complex project than ours) doesn't use it, we shouldn't use it as well. Friendly reminder that when we call `GC.SuppresFinilizer` we extend the life of that object.

### Solution

We shouldn't call the `Dispose` and/or `DisconnectHandler` so agressively, we should follow the same behavior as .NET MAUI , that way the user will experience the same behavior accross our lib and the MAUI framework.

### Drawbacks

This will be a breaking change, since the lifecycle of controls will change drastically causing issues and _fake memory leaks_. Sadly there's no docs or a guide on how devs should handle that (yet), as you can see there's [an open issue](https://github.com/dotnet/maui/issues/7381) to track it.

**Comment by CliffAgius** [comment]
Closed as per the Community Stand-up discussion here: https://youtu.be/N9wMcBP4jtg?t=2889
